### ISSUE TITLE
ubuntu16.04 build error benchmark

### ISSUE THREAD

**Issue by noobgrow** [facebookresearch/maskrcnn-benchmark#589]
`running build
running build_py
running build_ext
building 'maskrcnn_benchmark._C' extension
gcc -pthread -B /home/dong/anaconda3/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -DWITH_CUDA -I/home/dong/MASK_RCNN/maskrcnn-benchmark-master/mask/maskrcnn_benchmark/csrc -I/home/dong/anaconda3/lib/python3.6/site-packages/torch/lib/include -I/home/dong/anaconda3/lib/python3.6/site-packages/torch/lib/include/torch/csrc/api/include -I/home/dong/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH -I/home/dong/anaconda3/lib/python3.6/site-packages/torch/lib/include/THC -I/usr/local/cuda/include -I/home/dong/anaconda3/include/python3.6m -c /home/dong/MASK_RCNN/maskrcnn-benchmark-master/mask/maskrcnn_benchmark/csrc/vision.cpp -o build/temp.linux-x86_64-3.6/home/dong/MASK_RCNN/maskrcnn-benchmark-master/mask/maskrcnn_benchmark/csrc/vision.o -DTORCH_API_INCLUDE_EXTENSION_H -DTORCH_EXTENSION_NAME=_C -D_GLIBCXX_USE_CXX11_ABI=0 -std=c++11
cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
gcc -pthread -B /home/dong/anaconda3/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -DWITH_CUDA -I/home/dong/MASK_RCNN/maskrcnn-benchmark-master/mask/maskrcnn_benchmark/csrc -I/home/dong/anaconda3/lib/python3.6/site-packages/torch/lib/include -I/home/dong/anaconda3/lib/python3.6/site-packages/torch/lib/include/torch/csrc/api/include -I/home/dong/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH -I/home/dong/anaconda3/lib/python3.6/site-packages/torch/lib/include/THC -I/usr/local/cuda/include -I/home/dong/anaconda3/include/python3.6m -c /home/dong/MASK_RCNN/maskrcnn-benchmark-master/mask/maskrcnn_benchmark/csrc/cpu/nms_cpu.cpp -o build/temp.linux-x86_64-3.6/home/dong/MASK_RCNN/maskrcnn-benchmark-master/mask/maskrcnn_benchmark/csrc/cpu/nms_cpu.o -DTORCH_API_INCLUDE_EXTENSION_H -DTORCH_EXTENSION_NAME=_C -D_GLIBCXX_USE_CXX11_ABI=0 -std=c++11
cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
In file included from /home/dong/anaconda3/lib/python3.6/site-packages/torch/lib/include/ATen/ATen.h:9:0,
                 from /home/dong/anaconda3/lib/python3.6/site-packages/torch/lib/include/torch/csrc/api/include/torch/types.h:3,
                 from /home/dong/anaconda3/lib/python3.6/site-packages/torch/lib/include/torch/csrc/api/include/torch/data/dataloader_options.h:4,
                 from /home/dong/anaconda3/lib/python3.6/site-packages/torch/lib/include/torch/csrc/api/include/torch/data/dataloader.h:3,
                 from /home/dong/anaconda3/lib/python3.6/site-packages/torch/lib/include/torch/csrc/api/include/torch/data.h:3,
                 from /home/dong/anaconda3/lib/python3.6/site-packages/torch/lib/include/torch/csrc/api/include/torch/all.h:4,
                 from /home/dong/anaconda3/lib/python3.6/site-packages/torch/lib/include/torch/extension.h:4,
                 from /home/dong/MASK_RCNN/maskrcnn-benchmark-master/mask/maskrcnn_benchmark/csrc/cpu/vision.h:3,
                 from /home/dong/MASK_RCNN/maskrcnn-benchmark-master/mask/maskrcnn_benchmark/csrc/cpu/nms_cpu.cpp:2:
/home/dong/MASK_RCNN/maskrcnn-benchmark-master/mask/maskrcnn_benchmark/csrc/cpu/nms_cpu.cpp: In lambda function:
/home/dong/MASK_RCNN/maskrcnn-benchmark-master/mask/maskrcnn_benchmark/csrc/cpu/nms_cpu.cpp:71:47: error: invalid initialization of reference of type ‘const at::Type&’ from expression of type ‘c10::ScalarType’
   AT_DISPATCH_FLOATING_TYPES(dets.scalar_type(), "nms", [&] {
                                               ^
/home/dong/anaconda3/lib/python3.6/site-packages/torch/lib/include/ATen/Dispatch.h:15:32: note: in definition of macro ‘AT_DISPATCH_FLOATING_TYPES’
     const at::Type& the_type = TYPE;                                         \
                                ^
error: command 'gcc' failed with exit status 1
`
I use gcc4.9

pytorch-nightly-1.0.0.dev20181204-py3.6_cuda9.0.176_cudnn7.4.1_0.tar.bz2

how can I solve it?

**Comment by noobgrow** [comment]
Maybe its my torch install problem.
I solved it by reinstall pytorch-nightly.
I use the local install command as follows:
conda install --use-local pytorch-nightly-1.0.0.dev20190318-py3.6_cuda9.0.176_cudnn7.4.2_0.tar.bz2

**Comment by LeviViana** [comment]
You did the right thing. Please refer to #557.

**Comment by fmassa** [comment]
Closing folloing @LeviViana comment.
